.github/workflows/linux-publish.yml

```diff
@@ -0,0 +1,172 @@
+name: Publish Linux Packages
+
+on:
+  push:
+    branches: [ "main" ]
+    tags: 
+      # Publish semver tags as releases.
+      - app-ziti-console-v*.*.*
+
+jobs:
+  parse-version:
+    name: Find Console Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.parse.outputs.version }}
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - name: Derive version from tag or package
+        id: parse
+        shell: bash
+        env:
+          GITHUB_REF_NAME: ${{ github.ref_name }}
+          GITHUB_RUN_ID: ${{ github.run_id }}
+        run: |
+          set -o pipefail
+          if [[ "${GITHUB_REF_NAME}" == main ]]
+          then
+            if PACKAGE_VERSION=$(jq -r .version package.json)
+            then
+              echo "version=${PACKAGE_VERSION#v}-${GITHUB_RUN_ID}" | tee -a $GITHUB_OUTPUT
+            else
+              echo "ERROR: Could not find version in package.json"
+              exit 1
+            fi
+          else
+            echo "version=${GITHUB_REF_NAME#app-ziti-console-v}" | tee -a $GITHUB_OUTPUT
+          fi
+
+  build-and-upload-app-ziti-console:
+    needs: parse-version
+    name: Build Console ZIP
+    runs-on: ubuntu-latest
+    env:
+      BUILD_CONFIG: "Release"
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
```

Comment:
this node-version -- can we make it a constant somehow? i suppose not but it's also used in releases.yml in two locations. just wondering if it's possible

Reply:
If a GH repo admin sets a GH Actions variable named `NODE_VERSION` then we could do this:

```yaml
      - uses: actions/setup-node@v4
        with:
          node-version: ${{ vars.NODE_VERSION }}
```

Reply:
meh. that's not better for sure. thx